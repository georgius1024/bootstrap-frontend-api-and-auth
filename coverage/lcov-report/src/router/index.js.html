<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/router/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/router</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>16/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">43.75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>7/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/32</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">64x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Vue from 'vue'
import Router from 'vue-router'
import store from '@/store'
import eventBus from '@/event-bus'
import activator from './activator'
Vue.use(Router)
&nbsp;
function load (component) {
  return require(`@/views/${component}.vue`).default
}
&nbsp;
const ifNotAuthenticated = (to, from, next) =&gt; {
  if (!store.getters.isAuthenticated) {
    next()
    return
  }
  next('/')
}
&nbsp;
const ifAuthenticated = (to, from, next) =&gt; {
  if (store.getters.isAuthenticated) {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (store.getters['account/mustChangePassword']) {
<span class="cstat-no" title="statement not covered" >      eventBus.error('Необходимо установить новый пароль!')</span>
<span class="cstat-no" title="statement not covered" >      next('/reset-password')</span>
    } else {
      next()
    }
    return
  }
  next('/login')
}
const ifMustChangePassword = <span class="fstat-no" title="function not covered" >(t</span>o, from, next) =&gt; {
<span class="cstat-no" title="statement not covered" >  if (store.getters['account/mustChangePassword']) {</span>
<span class="cstat-no" title="statement not covered" >    next()</span>
<span class="cstat-no" title="statement not covered" >    return</span>
  }
<span class="cstat-no" title="statement not covered" >  next('/')</span>
}
&nbsp;
export default new Router({
  routes: [
    {
      path: '/',
      name: 'root',
      component: load('root'),
      beforeEnter: ifAuthenticated
    },
    {
      path: '/profile',
      name: 'profile',
      component: load('profile'),
      beforeEnter: ifAuthenticated
    },
    {
      path: '/login',
      name: 'login',
      component: load('login'),
      beforeEnter: ifNotAuthenticated
    },
    {
      path: '/register',
      name: 'register',
      component: load('register'),
      beforeEnter: ifNotAuthenticated
    },
    {
      path: '/change-password',
      name: 'change-password',
      component: load('change-password'),
      beforeEnter: ifAuthenticated
    },
    {
      path: '/reset-password',
      name: 'reset-password',
      component: load('reset-password'),
      beforeEnter: ifMustChangePassword
    },
    {
      path: '/about',
      name: 'about',
      component: load('about'),
      meta: {
        path: ['О приложении']
      }
    },
    {
      path: '/activate',
      name: 'activator',
      beforeEnter: <span class="fstat-no" title="function not covered" >as</span>ync (to, from, next) =&gt; {
        const success = <span class="cstat-no" title="statement not covered" >await activator(to.query['code'])</span>
<span class="cstat-no" title="statement not covered" >        if (success) {</span>
<span class="cstat-no" title="statement not covered" >          next('/')</span>
        } else {
<span class="cstat-no" title="statement not covered" >          next('/login')</span>
        }
      }
    },
    {
      path: '/logout',
      name: 'logout',
      beforeEnter: <span class="fstat-no" title="function not covered" >as</span>ync (to, from, next) =&gt; {
<span class="cstat-no" title="statement not covered" >        if (store.getters.isAuthenticated) {</span>
          const Api = <span class="cstat-no" title="statement not covered" >require('@/api').default</span>
<span class="cstat-no" title="statement not covered" >          await Api.get('/private/logout')</span>
<span class="cstat-no" title="statement not covered" >          store.commit('account/logout')</span>
        }
<span class="cstat-no" title="statement not covered" >        next('/login')</span>
      }
    },
    {
      path: '*',
      component: load('404')
    }
  ],
  scrollBehavior (to, from, savedPosition) {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (savedPosition &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >savedPosition.y)</span> {
<span class="cstat-no" title="statement not covered" >      return savedPosition</span>
    } else {
      return { x: 0, y: 0 }
    }
  }
})
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 16 2018 12:20:50 GMT+0500 (GMT+05:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
