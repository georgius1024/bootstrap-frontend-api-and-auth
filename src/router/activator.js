import config from '@/config'
import Api from '@/api'
import store from '@/store'

async function activator (code) {
  Api.setBaseUrl(config.API_URL)
  try {
    const response = await Api.get('auth/once/' + code)
    const user = response.data
    const auth = response.auth
    store.commit('account/login', { user, auth })
    return true
  } catch (error) {
    return false
  }
}
export default activator
