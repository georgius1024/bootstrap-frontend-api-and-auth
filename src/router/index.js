import Vue from 'vue'
import Router from 'vue-router'
import store from '@/store'
import eventBus from '@/event-bus'
import activator from './activator'
Vue.use(Router)

function load (component) {
  return require(`@/views/${component}.vue`).default
}

const ifNotAuthenticated = (to, from, next) => {
  if (!store.getters.isAuthenticated) {
    next()
    return
  }
  next('/')
}

const ifAuthenticated = (to, from, next) => {
  if (store.getters.isAuthenticated) {
    if (store.getters['account/mustChangePassword']) {
      eventBus.error('Необходимо установить новый пароль!')
      next('/reset-password')
    } else {
      next()
    }
    return
  }
  next('/login')
}
const ifMustChangePassword = (to, from, next) => {
  if (store.getters['account/mustChangePassword']) {
    next()
    return
  }
  next('/')
}

export default new Router({
  routes: [
    {
      path: '/',
      name: 'root',
      component: load('root'),
      beforeEnter: ifAuthenticated
    },
    {
      path: '/profile',
      name: 'profile',
      component: load('profile'),
      beforeEnter: ifAuthenticated
    },
    {
      path: '/login',
      name: 'login',
      component: load('login'),
      beforeEnter: ifNotAuthenticated
    },
    {
      path: '/register',
      name: 'register',
      component: load('register'),
      beforeEnter: ifNotAuthenticated
    },
    {
      path: '/change-password',
      name: 'change-password',
      component: load('change-password'),
      beforeEnter: ifAuthenticated
    },
    {
      path: '/reset-password',
      name: 'reset-password',
      component: load('reset-password'),
      beforeEnter: ifMustChangePassword
    },
    {
      path: '/about',
      name: 'about',
      component: load('about'),
      meta: {
        path: ['О приложении']
      }
    },
    {
      path: '/activate',
      name: 'activator',
      beforeEnter: async (to, from, next) => {
        const success = await activator(to.query['code'])
        if (success) {
          next('/')
        } else {
          next('/login')
        }
      }
    },
    {
      path: '/logout',
      name: 'logout',
      beforeEnter: async (to, from, next) => {
        if (store.getters.isAuthenticated) {
          const Api = require('@/api').default
          await Api.get('/private/logout')
          store.commit('account/logout')
        }
        next('/login')
      }
    },
    {
      path: '*',
      component: load('404')
    }
  ],
  scrollBehavior (to, from, savedPosition) {
    if (savedPosition && savedPosition.y) {
      return savedPosition
    } else {
      return { x: 0, y: 0 }
    }
  }
})
