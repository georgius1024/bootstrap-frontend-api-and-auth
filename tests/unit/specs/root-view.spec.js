import {
  shallowMount
} from '@vue/test-utils'

import router from '@/router'
import view from '@/views/root'

describe('root-view.vue', () => {
  it('should render correct contents ', async () => {
    const wrapper = shallowMount(view, {
      router
    })
    expect(wrapper).toMatchSnapshot()
  })
})
