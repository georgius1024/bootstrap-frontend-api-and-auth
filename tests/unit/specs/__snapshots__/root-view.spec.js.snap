// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root-view.vue should render correct contents  1`] = `
<centeredlayout-stub>
  <vscaletransition-stub origin="top center 0">
    <vflex-stub tag="div" xs12="" sm8="" lg6="" class="my-4">
      <vcard-stub tag="div" color="secondary lighten-5">
        <vcardtext-stub class="px-5 pb-5"><img src="@/assets/logo.png" class="logo hidden-md-and-down">
          <p class="hidden-md-and-down text-xs-center">
            AWESOME FRONTEND APP

          </p>
          <vdivider-stub class="my-2 hidden-md-and-down"></vdivider-stub>
          <vlayout-stub tag="div" column="">
            <vbtn-stub activeclass="v-btn--active" block="true" large="true" tag="button" type="button" disabled="" class="primary">
              Action 1
            </vbtn-stub>
            <vbtn-stub activeclass="v-btn--active" block="true" large="true" tag="button" type="button" disabled="" class="primary">
              Action 2
            </vbtn-stub>
            <vbtn-stub activeclass="v-btn--active" block="true" large="true" tag="button" type="button" disabled="" class="primary">
              Action 3
            </vbtn-stub>
            <vbtn-stub activeclass="v-btn--active" block="true" large="true" tag="button" type="button" disabled="" class="primary">
              Action 4
            </vbtn-stub>
          </vlayout-stub>
        </vcardtext-stub>
      </vcard-stub>
    </vflex-stub>
  </vscaletransition-stub>
</centeredlayout-stub>
`;
